src/Engine/ProtoCore/Core.cs

```diff
@@ -214,7 +214,7 @@ public struct ErrorEntry
         public List<CodeBlock> CodeBlockList { get; set; }
         // The Complete Code Block list contains all the code blocks
         // unlike the codeblocklist which only stores the outer most code blocks
-        public List<CodeBlock> CompleteCodeBlockList { get; set; }
+        public SortedDictionary<int, CodeBlock> CompleteCodeBlockList { get; set; }
```

Comment:
@pinzart I don't think we can do this. Unfortunately, lot's of code in here appears to be unnecessarily public. Even though we are in the "engine" we do not know who is using these public members, and we have stated publicly we will follow semantic versioning.

It's entirely possible an extension *could* access these data structures.

What about obsoleting this member, having it return the values inside the new dictionary, and adding a new one - and lastly consider if the new member should actually be public.

Reply:
hmm..I had hoped the VM was not avavilablr to api users.
But I se you are correct, I updated the code. will add obsolete stmts